eslint/eslint-plugin/src/test/typedef-var.test.ts

```diff
@@ -46,6 +46,12 @@ ruleTester.run('typedef-var', typedefVar, {
     {
       code: 'for (const x of []) { }'
     },
+    {
+      code: 'const x = 1 as const;'
+    },
+    {
+      code: 'const x: number = 1;'
```

Comment:
```suggestion
      code: 'const x: 1 = 1;'
```
Is what TypeScript would normally infer, so this should really be the reference case.

Reply:
If you say `const x = 1;`, the TypeScript compiler will infer the type of `x` to be `1`. That is the most specific type that `x` can be (because it's a const, so it can never be any other number or other value).

Take a look in the [TypeScript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAHjAvDAjAbgFBA). If you mouse-over the `x`, IntelliSense will tell you what the inferred type is:
![image](https://github.com/microsoft/rushstack/assets/5010588/e0c37f18-4358-4818-8b29-3e9e8bb5b23c)

Although I kinda agree that we should have a second example case of `const x: number = 1;`.

Reply:
yeah, I understood the part about type inference in the comment, I'm just not sure what are the actual requested changes in this case and why
anyway, added a second test case

Reply:
@dmichon-msft prefers the `const x: 1 = 1;` syntax to `const x: number = 1;` because it's more strict, and I've come to agree, so I think that's why he requested the change. It's a change we'd recommend someone make in "real" code in a PR.